packages/xo-server/src/xo-mixins/remotes.mjs

```diff
@@ -172,11 +174,16 @@ export default class {
             })
           : this.getRemoteHandler(remote.id).then(handler => handler.getInfo())
 
+      const totalBackupSize = await Disposable.use(this._app.getBackupsRemoteAdapter(remote), adapter =>
```

Comment:
What do you think of that @julien-f?

Reply:
I agree that this call should be as fast as possible, and it should absolutely not block when something it's not responding.

This is a problem we already have in XO at a few places:

- https://github.com/vatesfr/xen-orchestra/blob/530c30fd50c62a5b949b755768642b076f092d99/packages/xo-server/src/api/remote.mjs#L18-L20
- https://github.com/vatesfr/xen-orchestra/blob/530c30fd50c62a5b949b755768642b076f092d99/packages/xo-server/src/api/backup-ng.mjs#L266-L268

I see a couple of solutions to work-around this problem:

1. split the request so that each request is tied to a resource to avoid a single one from blocking everything (but that's not what we want for the dashboard)
2. fetch the data asynchronously and invalidate the global cache when some new data is available

IMHO, the second solution is a bit more complex but it's likely the best one.

What do both of you think?

Reply:
I think, that , as long s we don't use them outside of the dashboard,  it is acceptable that `getTotalBackupSize`  is not included in the remote info, but called specifically 

On the long run, I think we should avoid listing too many files on the remote, and get most of the data from a local cache/database, updated on write.  That implies not having concurrent XO writing on the same remote, and having the merge worker inform the XO of it's job

Reply:
We decide to merge the PR as is and to create a new one for detached backup size computation.